src/WebJobs.Script/Description/NotificationHubBindingMetadata.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.NotificationHubs;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    internal class NotificationHubBindingMetadata : BindingMetadata
+    {
+        public string TagExpression { get; set; }
+
+        [AllowNameResolution]
+        public string ConnectionString { get; set; }
+
+        [AllowNameResolution]
+        public string HubName { get; set; }
+
+        public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
+        {
+            if (configBuilder == null)
+            {
+                throw new ArgumentNullException("configBuilder");
+            }
+
+            NotificationHubConfiguration config = new NotificationHubConfiguration();
+            if (!string.IsNullOrEmpty(ConnectionString) && !string.IsNullOrEmpty(HubName))
+            {
+                config.ConnectionString = ConnectionString;
+                config.HubName = HubName;
+            }
```

Comment:
change this to two separate if blocks -- it's possible that ConnectionString is set in the config yet HubName is not (or vice versa)